### PR TITLE
LAM-938: Add tron and trc-20 usdt 

### PR DESCRIPTION
**Pull request by RafaelTaranto**

### CODE REVIEW

File: lib/plugins/wallet/trongrid/trongrid.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+const base = require('../tron/base')
+
+const NAME = 'trongrid'
+
+function run (account) {
+  if (!account.endpoint) throw new Error('Need to configure API endpoint for trongrid')
+
+  const endpoint = 'https://api.trongrid.io'
+
+  base.connect({ ...account, endpoint })
```
Comment (siiky) [inline lib/plugins/wallet/trongrid/trongrid.js]: Shouldn't `account.endpoint` be used as the endpoint?

File: lib/plugins/exchange/binance.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ const { ORDER_TYPES } = require('./consts')
 
 const ORDER_TYPE = ORDER_TYPES.MARKET
 const { BTC, BCH, XMR, ETH, LTC, ZEC } = COINS
-const CRYPTO = [BTC, ETH, LTC, ZEC, BCH, XMR]
+const CRYPTO = [BTC, ETH, LTC, ZEC, BCH, XMR ]
```
Comment (siiky) [inline lib/plugins/exchange/binance.js]: `USDT` missing?

File: lib/plugins/wallet/tron/base.js [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+  if (!mnemonic) throw new Error('No mnemonic seed!')
+
+  return TronWeb.fromMnemonic(mnemonic.replace(/[\r\n]/gm, ' ').trim(), `${DEFAULT_PREFIX_PATH}\/0`)
+}
+
+function paymentWallet (account, index) {
+  const mnemonic = account.mnemonic
+  if (!mnemonic) throw new Error('No mnemonic seed!')
+
+  return TronWeb.fromMnemonic(mnemonic.replace(/[\r\n]/gm, ' ').trim(), `${PAYMENT_PREFIX_PATH}\/${index}`)
```
Comment (siiky) [inline lib/plugins/wallet/tron/base.js]: Should `\/` after `PAYMENT_PREFIX_PATH` be escaped?

File: lib/plugins/wallet/tron/base.js [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+}
+
+function getStatus (account, tx, requested, settings, operatorId) {
+  const { toAddress, cryptoCode } = tx
+  return checkCryptoCode(cryptoCode)
+    .then(code => confirmedBalance(toAddress, code))
+    .then((confirmed) => {
+      if (confirmed.gte(requested)) return { receivedCryptoAtoms: confirmed, status: 'confirmed' }
+      return { receivedCryptoAtoms: 0, status: 'notSeen' }
+    })
```
Comment (siiky) [inline lib/plugins/wallet/tron/base.js]: Isn't it possible to get a transaction of part of the requested value? Like with the other wallets:

https://github.com/lamassu/lamassu-server/blob/e3892e9ac2f111fedf26545441c18419250cb773/lib/plugins/wallet/bitcoind/bitcoind.js#L148

**Review by siiky — CHANGES_REQUESTED** [review]

File: lib/plugins/wallet/trongrid/trongrid.js (hunk 1 above)
Comment (RafaelTaranto) [reply lib/plugins/wallet/trongrid/trongrid.js]: It was decided this is hardcoded, new version doesn't have the check.

File: lib/plugins/exchange/binance.js (hunk 2 above)
Comment (RafaelTaranto) [reply lib/plugins/exchange/binance.js]: USDT doesn't work for this one, I'll remove the empty space

File: lib/plugins/wallet/tron/base.js (hunk 3 above)
Comment (RafaelTaranto) [reply lib/plugins/wallet/tron/base.js]: it's already an escaped forward slash

File: lib/plugins/wallet/tron/base.js (hunk 4 above)
Comment (RafaelTaranto) [reply lib/plugins/wallet/tron/base.js]: Good catch, updated.

Comment (RafaelTaranto) [reply lib/plugins/wallet/tron/base.js]: Actually the problem here is a different one, we don't really have pending amount. Same thing happens with ETH so I'll get confirmation from tech first to see what to do here.

**Pull request merged**